sdk/src/client/api/block_builder/input_selection/mod.rs

```diff
@@ -160,10 +160,7 @@ impl InputSelection {
         self.available_inputs
             .retain(|input| !self.forbidden_inputs.contains(input.output_id()));
 
-        // This is to avoid a borrow of self since there is a mutable borrow in the loop already.
-        let required_inputs = std::mem::take(&mut self.required_inputs);
-
-        for required_input in required_inputs {
+        for required_input in self.required_inputs.clone() {
```

Comment:
Why not keep the `take` since we're not using `required_inputs` anymore?

Reply:
We are using it now, in `initial_mana_excess`

Reply:
Ok so for my understanding, the compiler wouldn't say anything if we take and then reuse later?

Reply:
no because take leaves an empty vec